File: src/main/java/cz/stasimek/fakturaceeasypeasy/enumeration/Provider.java
package cz.stasimek.fakturaceeasypeasy.enumeration;

import cz.stasimek.fakturaceeasypeasy.exception.ApplicationException;

public enum Provider {
	GITHUB,
	GOOGLE;

	public static Provider valueOfCaseInsensitive(String search) {
		for (Provider provider : values()) {
			if (provider.name().compareToIgnoreCase(search) == 0) {
				return provider;
			}
		}
		throw new ApplicationException(
				String.format("Provider %s not supported.", search)
		);
	}
}

File: src/main/kotlin/cz/stasimek/fakturaceeasypeasy/enumeration/Provider.kt
package cz.stasimek.fakturaceeasypeasy.enumeration

import cz.stasimek.fakturaceeasypeasy.exception.ApplicationException

enum class Provider {
    GITHUB, GOOGLE;

    companion object {
        fun valueOfCaseInsensitive(search: String): Provider {
            for (provider in values()) {
                if (provider.name.compareTo(search, ignoreCase = true) == 0) {
                    return provider
                }
            }
            throw ApplicationException(String.format("Provider %s not supported.", search))
        }
    }
}